File: src/main/java/com/weeth/global/config/WebMvcConfig.java
package com.weeth.global.config;

import com.weeth.global.auth.jwt.service.JwtService;
import com.weeth.global.auth.resolver.CurrentUserArgumentResolver;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.method.support.HandlerMethodArgumentResolver;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.util.List;

@Configuration
@RequiredArgsConstructor
public class WebMvcConfig implements WebMvcConfigurer {

    private final JwtService jwtService;

    @Override
    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
        resolvers.add(new CurrentUserArgumentResolver(jwtService));
    }
}

File: src/main/kotlin/com/weeth/global/config/WebMvcConfig.kt
package com.weeth.global.config

import com.weeth.global.auth.resolver.CurrentUserArgumentResolver
import com.weeth.global.auth.resolver.CurrentUserRoleArgumentResolver
import org.springframework.context.annotation.Configuration
import org.springframework.web.method.support.HandlerMethodArgumentResolver
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebMvcConfig : WebMvcConfigurer {
    override fun addArgumentResolvers(resolvers: MutableList<HandlerMethodArgumentResolver>) {
        resolvers.add(CurrentUserArgumentResolver())
        resolvers.add(CurrentUserRoleArgumentResolver())
    }
}
